src/lib/y2users/autoinst/reader.rb

```diff
@@ -99,6 +105,41 @@ def create_password(user)
         password.account_expiration = AccountExpiration.new(password_settings.expire)
         password
       end
+
+      # Validates the user and adds the problems found to the given issues list
+      #
+      # @param user_section [Y2Users::AutoinstProfile::UserSection] User section from the profile
+      # @param issues [Y2Issues::List] Add issues list
+      # @return [Boolean] true if the user is valid; false otherwise.
+      def valid_user?(user_section, issues)
```

Comment:
NP: Not sure if I like these boolean methods called `valid_bla?` that, in addition to returning true or false, populate the given issues list. But I guess that's something we will fix with https://trello.com/c/yVrdlyCx/4747-refactor-linuxwriter (I added a paragraph about having the readers also into account).

Reply:
Well, I am not a big fan, but it is a private API so I am fine with it. Additionally, a user can be valid although it contains issues.